publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/identifiers/HandleIdentifierEventHandler.java

```diff
@@ -0,0 +1,175 @@
+                               .getResponsePayload();
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .forEach(this::processInputPayload);
+        return null;
+    }
+
+    private CognitoCredentials fetchCredentials() {
+        var credentials = secretsManagerClient.fetchClassSecret(backendClientSecretName,
+                                                                BackendClientCredentials.class);
+        var uri = UriWrapper.fromHost(backendClientAuthUrl).getUri();
+        return new CognitoCredentials(credentials::getId, credentials::getSecret, uri);
+    }
+
+    protected Void processInputPayload(EventReference input) {
+        var eventBlob = s3Driver.readEvent(input.getUri());
+
+        if (RESOURCE_UPDATE_EVENT_TOPIC.equals(input.getTopic())) {
+            var resourceUpdate = parseResourceUpdateInput(eventBlob);
+            if (isPublished(resourceUpdate) && isMissingHandle(resourceUpdate)) {
+                logger.info("Creating handle for publication: {}", resourceUpdate.getIdentifier());
+                var userInstance = UserInstance.create(resourceUpdate.getOwner(), resourceUpdate.getCustomerId());
+                var publication = fetchPublication(userInstance, resourceUpdate.getIdentifier());
+                var additionalIdentifiers = new HashSet<>(publication.getAdditionalIdentifiers());
+                var handle = createNewHandle(getLandingPage(publication.getIdentifier()));
+                logger.info("Created handle: {}", handle.value());
+                additionalIdentifiers.add(handle);
+                publication.setAdditionalIdentifiers(additionalIdentifiers);
+                attempt(() -> resourceService.updatePublication(publication));
+            }
+        }
+        return null;
+    }
+
+    private URI getLandingPage(SortableIdentifier identifier) {
+        return new UriWrapper(HTTPS, frontendDomain).addChild(REGISTRATION_PATH)
+                   .addChild(identifier.toString())
+                   .getUri();
+    }
+
+    private static boolean isMissingHandle(Resource resourceUpdate) {
+        return !resourceContainsHandle(resourceUpdate) && !resourceContainsLegacyHandle(resourceUpdate);
+    }
+
+    private static boolean resourceContainsHandle(Resource resourceUpdate) {
+        return resourceUpdate.getAdditionalIdentifiers().stream().anyMatch(HandleIdentifier.class::isInstance);
+    }
+
+    private static boolean resourceContainsLegacyHandle(Resource resourceUpdate) {
+        return resourceUpdate.getAdditionalIdentifiers()
+                   .stream()
+                   .filter(AdditionalIdentifier.class::isInstance)
+                   .anyMatch(a -> LEGACY_HANDLE_SOURCE_NAME.equals(a.sourceName()));
+    }
+
+    private static boolean isPublished(Resource resourceUpdate) {
+        return PUBLISHED_STATUSES.contains(resourceUpdate.getStatus());
```

Comment:
I take whatever publication that is changed in any way and check if its published (now or from before, does not matter), and if its missing handle, I will give it one. I was now just thinking about how this will work when importing from cristin and then later brage. Its easy to get double handles then, I will talk with JE

Reply:
Solved by disabling event rule. Should not be enabled before ALL migrations are done.